astropy/io/fits/tests/test_convenience.py

```diff
@@ -157,6 +157,28 @@ def test_tabledump(self):
         fits.tabledump(self.temp('tb.fits'), datafile=self.temp('test_tb.txt'))
         assert os.path.isfile(self.temp('test_tb.txt'))
 
+    def test_append(self):
+        """
+        Test that append works in the case where the file does not exist or is empty, and
+        also the case in which ``verify`` is True or if ``filename`` is a file object.
+        """
+        # Test case 1
+        fits.append('test_out_1.fits', data=np.ones((16,32)), output_verify='fix')
+        with fits.open('test_out_1.fits', checksum=True, verify=True) as hdu1:
+            assert isinstance(hdu1[0], fits.PrimaryHDU)
+            assert np.all(hdu1[0].data == np.ones((16,32)))
+            assert hdu1[0].header is not False
```

Comment:
Please keep only the data test. `.header` will never be False, and the first HDU will always be a primary one.

Reply:
Yup, removed.